src/libraries/System.IO.Compression/tests/ZipArchive/zip_ReadTests.cs

```diff
@@ -273,6 +273,37 @@ public static async Task EnsureDisposeIsCalledAsExpectedOnTheUnderlyingStream(bo
             Assert.Equal(expectedDisposeCalls, disposeCallCountingStream.NumberOfDisposeCalls);
         }
 
+        [Theory]
+        // A 19-character filename will result in a 65-byte central directory header. 64 of these will make the central directory
+        // read process stretch into two 4KB buffers.
+        [InlineData(64, "example/file-{0:00}.dat")]
+        public static void CanReadLargeCentralDirectoryHeader(int count, string entryNameFormat)
+        {
+            using (MemoryStream archiveStream = new MemoryStream())
```

Comment:
If there's only one `InlineData`, we can convert this to a `Fact`.

```suggestion
        // A 19-character filename will result in a 65-byte central directory header. 64 of these will make the central directory
        // read process stretch into two 4KB buffers.
        [Fact]
        public static void CanReadLargeCentralDirectoryHeader()
        {
            int count = 64;
            string entryNameFormat = "example/file-{0:00}.dat";

            using (MemoryStream archiveStream = new MemoryStream())
```